src/main/java/javax/validation/Validation.java

```diff
@@ -380,7 +387,30 @@ public ValidationProviderResolver getDefaultValidationProviderResolver() {
 		}
 
 		private synchronized void cacheValidationProviders(ClassLoader classLoader, List<ValidationProvider<?>> providers) {
-			providersPerClassloader.put( classLoader, new SoftReference<List<ValidationProvider<?>>>( providers ) );
+			providersPerClassloader.put( classLoader, new SoftReference<>( providers ) );
+		}
+	}
+
+	private static class NewProviderInstance<T extends ValidationProvider<?>> implements PrivilegedAction<T> {
+
+		private final Class<T> clazz;
+
+		public static <T extends ValidationProvider<?>> NewProviderInstance<T> action(Class<T> clazz) {
+			return new NewProviderInstance<>( clazz );
+		}
+
+		private NewProviderInstance(Class<T> clazz) {
```

Comment:
We fail even faster than that. See https://github.com/gunnarmorling/beanvalidation-api/blob/61f27224044b33beaa6677a06cacda577208719c/src/main/java/javax/validation/Validation.java#L198 .

As the usage of this class is very specific, I don't think it's worth it to add another null test.